plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java

```diff
@@ -151,6 +155,22 @@ public void testInsertWriteBulkinessWithTimestamps(String timestampType)
         }
     }
 
+    @Test(dataProvider = "testTableNameDataProvider")
+    public void testCreateTableWithSpecialCharacterName(String tableName)
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_TABLE)) {
+            assertQueryFails("CREATE TABLE " + "\"" + tableName + "\"" + " (a bigint, b double, c varchar(50))", "This connector does not support creating tables");
+            return;
+        }
+
+        assertUpdate("CREATE TABLE " + "\"" + tableName + "\"" + " (a bigint, b double, c varchar(50))");
+        assertTrue(getQueryRunner().tableExists(getSession(), tableName.replace("\"\"", "\"")));
```

Comment:
`.replace("\"\"", "\"")` looks fishy here.
`QueryRunner.tableExists()` should be given a table name as it is, with no "SQL-like formatting" or any other escapes applied

are you working around some bug?

Reply:
Yep it's kind of workaround because for this test : 
`.add("tri\"\"cky")`

When we create table using trino if we want create table with such name:` tri"cky `we need to pass to trino double double quotes like `tri""cky` in another way trino parser will not pass it and then it replace double double quotes with double quotes `"" -> "` so final table name in Database will be ` tri"cky ` as we want. But when we query this table with : 
`assertTrue(getQueryRunner().tableExists(getSession(), tableName));`

We need to pass strict table name which we expect because it's not going through the trino parser engine it's kind of checking table metadata, so it expect strict column name with one double quote

Reply:
in BCT we have smth like this, I think It could help also in my case:
private static String toColumnNameInSql(String columnName, boolean delimited)
```
    private static String toColumnNameInSql(String columnName, boolean delimited)
    {
        String nameInSql = columnName;
        if (delimited) {
            nameInSql = "\"" + columnName.replace("\"", "\"\"") + "\"";
        }
        return nameInSql;
    }
```

Reply:
Agreed, we need to distinguish "actual table name" and "table name in SQL" in this test.